Tasks/SonarQubePostTest/PRCA/PostComments-Module.psm1

```diff
@@ -65,7 +65,7 @@ $script:messageToCommentListMap = $null
 # Comment source is used to decorate comments created by this logic. Only comments with the same source will be resolved.
 function PostAndResolveComments
 {
-    param ([Array][ValidateNotNull()]$messages, [string][ValidateNotNullOrEmpty()]$messageSource)
+    param ([Array]$messages, [string][ValidateNotNullOrEmpty()]$messageSource)
```

Comment:
In Powershell an empty array is in some case interpreted as null. So I removed all the ValidateNotNull checks. It is possible to call PostAndResolveComments withouth any messages, i.e. just to resolve existing comments.